sdks/python/apache_beam/ml/inference/utils.py

```diff
@@ -125,17 +125,21 @@ def __init__(
     """
     Watches a directory for updates to files matching a given file pattern.
 
-    **Note**: Start timestamp will be defaulted to timestamp when pipeline
-      was run. All the files matching file_pattern, that are uploaded before
-      the pipeline started will be discarded.
-
       Args:
-        file_pattern: A glob pattern used to watch a directory for model
-          updates.
+        file_pattern: The file path to read from as a local file path or a
+        GCS ``gs://`` path. The path can contain glob characters
+        (``*``, ``?``, and ``[...]`` sets).
         interval: Interval at which to check for files matching file_pattern
           in seconds.
         stop_timestamp: Timestamp after which no more files will be checked.
 
+    Constraints:
+    1. If the file is read and then there is an update to that file, this
+      transform will ignore that update. Always update a file with unique
+      name.
```

Comment:
This doesn't quite convey that the following won't work:

```
- Create model with name A (model A is now used)
- Create model with name B (model B is now used)
- Delete file with name A
- Create new model with name A (model A is still not used because of stateful DoFn)
```

Could we update to something like: `Any previously used filenames cannot be reused. If a file is added or updated to a previously used filename, this transform will ignore that update. To trigger a model update, always upload a file with unique name.`

(tried to keep as much of your wording as possible, feel free to reword if you don't like my phrasing).

Reply:
Changed the wording